#==================================================================================================
# Upstream bug fix, removing invalid cast to GCompareFunc. Fixed in gtk3 and later, but never
# backported to gtk2.
#
# Date: 2024-04-24
# See: https://trac.macports.org/ticket/69471
#==================================================================================================
--- gtk/gtkscale.c.orig	2024-04-24 14:08:36.000000000 -0400
+++ gtk/gtkscale.c	2024-04-24 14:09:30.000000000 -0400
@@ -1471,7 +1471,7 @@
   mark->position = position;
  
   priv->marks = g_slist_insert_sorted_with_data (priv->marks, mark,
-                                                 (GCompareFunc) compare_marks,
+                                                 compare_marks,
                                                  GINT_TO_POINTER (
                                                    gtk_range_get_inverted (GTK_RANGE (scale)) 
                                                    ));
